### DISCUSSION TITLE
指定d3d11-adapter并不完全等效控制面板设置以独显运行mpv

### DISCUSSION THREAD

**Discussion by nanachisol** [dyphire/mpv-config#120] [Q&A] [answered]
使用 I+N 的双显笔记本的内屏，系统为混合输出模式
1.在mpv.conf启用d3d11-adapter设置时，控制台提示Failed to query swap chain's output information，但是播放一切正常，无法分辨异常。
2.当在控制面版或者nvidia控制面板，指定mpv为独显运行，控制台输出一切正常。
此外右键菜单切换解码方式为nvdec，有办法可以同步切换后端gpu-api=vulkan或者opencl吗？
最后作者可以考虑在配置文件中加入Intel VSR and NVIDIA RTX scaling modes吗?

[mpv.log](https://github.com/user-attachments/files/17089340/mpv.log)

**Comment by nanachisol** [comment]
> [SSIM]
>  profile-desc=适用于 4k 低性能
>  glsl-shaders="~~/shaders/other/FastBilateral.glsl;~~/shaders/igv/SSimSuperRes.glsl;~~/shaders/igv/SSimDownscaler.glsl

FastBilateral.glsl在着色器文件夹中并不存在

**Reply by dyphire** [reply]
hh，遗留未更改，等下修复

**Comment by dyphire** [comment, marked as answer]
> 使用 I+N 的双显笔记本的内屏，系统为混合输出模式
1.在 mpv.conf 启用 d3d11-adapter 设置时，控制台提示 Failed to query swap chain's output information，但是播放一切正常，无法分辨异常。

无法复现，我特意尝试了`d3d11-adapter="Intel(R) UHD Graphics"`指定核显，日志输出正常
```
[   0.882][v][vo/gpu-next/d3d11] Initializing GPU context 'd3d11'
[   0.978][v][vo/gpu-next/d3d11] Using Direct3D 11 feature level 12_1
[   0.979][v][vo/gpu-next/d3d11] Device Name: Intel(R) UHD Graphics
[   0.979][v][vo/gpu-next/d3d11] Device ID: 8086:9a60 (rev 01)
[   0.979][v][vo/gpu-next/d3d11] Subsystem ID: 1d05:1146
[   0.979][v][vo/gpu-next/d3d11] LUID: 00000000000171cc
```
再改为指定 N 卡也一样正常

> 2.当在控制面版或者 nvidia 控制面板，指定 mpv 为独显运行，控制台输出一切正常。
此外右键菜单切换解码方式为 nvdec，有办法可以同步切换后端 gpu-api=vulkan 或者 opencl 吗？

想要同时切换只需要自行指定相应的组合就行，新版 mpv 支持运行时更新 gpu-api，例如：
```
key   set gpu-api vulkan;set hwdec nvdec
```

> 最后作者可以考虑在配置文件中加入 Intel VSR and NVIDIA RTX scaling modes 吗?

无论是 Intel VSR 还是 NVIDIA RTX SR 的效果甚至不如 Anime4k 着色器，而 Anime4k 的效果就已经无法让我认可了，何况它们。这种所谓的超分不会被集成到本配置中，如果一定要尝试，自行研究 d3d11vpp 滤镜的用法，不提供支持

**Reply by nanachisol** [reply]
感谢，修改input.conf之后，mpv可以在切换nvdec时，自动启用vulkan后端。
关于Failed to query swap chain's output information，如这个issue所提及，(https://github.com/bottlesdevs/Bottles/issues/3268)
 可能与nv驱动有关。ps:更新驱动后并不能修正
最后，针对真人影片upscaling，有什么着色器或组合推荐吗？

**Reply by dyphire** [reply]
> 最后，针对真人影片upscaling，有什么着色器或组合推荐吗？

实拍类的 upscaling 基本没什么要选的，它和动画有很大不同
动画的画面基本都由线条构成而且相对简单，当使用对线条进行锐化处理的 ArtCNN、FSR 和 Anime4k 等升频算法时会带来明显观感提升（但对标清内容使用时会有明显的劣化
实拍类则不同，实拍类的画面信息要复杂的多且非由线条构成。这使得对实拍类应用锐化算法只会引入劣化效果而很难产生观感上的提升
实拍类的内容使用类似 ewa_hanning 质量的升频算法已经足够，视情况可以加入降噪类着色器减轻重噪。没必要折腾实拍类的升频算法因为很难产生肉眼可见的提升

**Comment by nanachisol** [comment]
现在通过配置文件指定显卡，mpv的控制台不会出现黄字警告了。
不清楚是系统还是显卡或者mpv更新的原因，最终解决了这个问题。
edit:不过在系统内指定显卡和在mpv配置文件中进行设置，效果还是不同的。
笔记本混合模式时候，是解码与渲染相分离的，在设定hwdec=vulkan-copy情况下，程序会闪退。
